spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/feign/FeignClientFactoryBean.java

```diff
@@ -128,6 +124,15 @@ public void setApplicationContext(ApplicationContext context) throws BeansExcept
 		return builder;
 	}
 
+	private Logger getLogger(FeignContext context) {
+		Logger logger = getOptional(context, Logger.class);
```

Comment:
Why are you doing this?  Why not just get the `FeignLoggerFactory` and create the logger?